solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -856,6 +912,11 @@ contract LiquidityProtection is ILiquidityProtection, TokenHandler, Utils, Owned
         uint256 _poolAmount,
         uint256 _reserveAmount
     ) internal returns (uint256) {
+        // notify event subscribers
+        if (address(eventsSubscriber) != address(0)) {
```

Comment:
called before provider pool is set - does it not need to know the provider has liquidity in that pool at this point?

Reply:
It doesn't, since its major goal is to only "sync" the previous state